cfme/tests/containers/test_projects_dashboard.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+
+from cfme.containers.provider import ContainersProvider
+from cfme.containers.container import Container
+from cfme.containers.image import Image
+from cfme.containers.service import Service
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.ignore_stream('5.8'),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+PROJECT_NAME = 'test-project-dashboard'  # Predefine project for this test
+
+
+@pytest.fixture(scope='function')
+def container_project_instance(appliance, provider):
+    collection_object = appliance.collections.container_projects
+    return collection_object.instantiate(name=PROJECT_NAME, provider=provider)
+
+
+def get_api_object_counts(appliance, project_name, provider):
+    """ Fetches amount of Containers/Services/Images from the API per selected project name"""
+    assert isinstance(provider, ContainersProvider)
+    return {
+        Container: len(provider.mgmt.list_container(project_name=project_name)),
+        Service: len(provider.mgmt.list_service(project_name=project_name)),
+        Image: len(get_container_images_amt(provider, project_name))
+    }
+
+
+def get_container_images_amt(provider, project_name=None):
+    """ Fetches images amount from the API per selected project name"""
+    images = provider.mgmt.list_image()
+    project_images = []
```

Comment:
For future reference, I think this could have been a generator in-line above. I've expanded it to 4 lines here for readability.

```python
project_images = [
    img for img 
    in provider.mgmt.list_image() 
    if img.image_project_name == project_name
]

Reply:
change the code, indeed that is more readable.